src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let stateClone = { ...state };
+  const transformations = [];
+
+  for (const action of actions) {
+    const transformedState = { ...stateClone };
+
+    if (action.type === 'addProperties') {
+      Object.assign(transformedState, action.extraData);
+    }
+
+    if (action.type === 'removeProperties') {
+      for (const property of action.keysToRemove) {
+        delete transformedState[property];
+      }
+    }
+
+    if (action.type === 'clear') {
+      for (const key in transformedState) {
+        delete transformedState[key];
+      }
+    }
```

Comment:
better to use a switch case here, 'cause you check everywhere action.type values